go-controller/pkg/ovn/gateway_init.go

```diff
@@ -495,11 +495,13 @@ func addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfAddr *net.IPNet) er
 	}
 
 	if config.Gateway.Mode == config.GatewayModeLocal {
-		ipMask := hostIfAddr.IP.Mask(hostIfAddr.Mask)
-		hostNet := &net.IPNet{IP: ipMask, Mask: hostIfAddr.Mask}
+		var matchDst string
 		// Local gw mode needs to use DGP to do hostA -> service -> hostB
-		matchStr = fmt.Sprintf("%s.src == %s && %s.dst == %s /* inter-%s */",
-			l3Prefix, mgmtPortIP, l3Prefix, hostNet.String(), nodeName)
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
```

Comment:
just to understand it, because the docs says that the OVN pipeline is (routing-> policy -> nat)
- we add a policy for all traffic coming from mgmtIP to ClusterSubnets, 
- the original traffic is mgmtIP to serviceIP, so it nor will be mgmtIP to clusterSubnet traffic until goes through NAT
or the load balancers does not work with nat in ovn?

Reply:
The load balancer (DNAT) is on the switch. So the service-> pod DNAT happens before the traffic ever gets to the router. Your first point is correct, this is for traffic coming from mp0 (the host) that is destined out of ovn network (needs to go back to the host via dgp)